File: cpp/n492_DesignSquare.cpp
// https://leetcode-cn.com/problems/construct-the-rectangle/
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <math.h>
using namespace std;

class Solution {
public:
    vector<int> constructRectangle(int area) {
        int w = sqrt(area);
        while(area % w != 0) w -= 1;
        int l = area / w;
        vector<int> res = {l, w};
        return res;
    }
};

int main(){
    return 0;
}

File: cpp/n492_DesignSquare.py
# https://leetcode-cn.com/problems/construct-the-rectangle/
class Solution:
    def constructRectangle(self, area):
        w = int(sqrt(area))
        while w > 0:
            if area % w == 0:
                l = int(area / w)
                break
            w = w - 1
        return [l, w] 
        
        
if __name__ == "__main__":
    pass